### ISSUE TITLE
Runtimelab_578/arr fails in crossgen2.

### ISSUE THREAD

**Issue by sandreenko** [dotnet/runtime#48204]
The test added in https://github.com/dotnet/runtime/pull/47400 is failing in crossgen2 outerloop with:
```
  Regressions/coreclr/Runtimelab_578/arr/arr.sh [FAIL]
      cp: target '/home/helixbot/work/AE900994/p/lib*.dylib' is not a directory
      rm: cannot remove '/home/helixbot/work/AE900994/w/B29009C8/e/Regressions/coreclr/Runtimelab_578/arr/composite-r2r.dll.rsp': No such file or directory
      Unhandled exception. System.AggregateException: One or more errors occurred. (Code generation failed for method '[arr]<Module>.main()')
       ---> ILCompiler.CodeGenerationFailedException: Code generation failed for method '[arr]<Module>.main()'
       ---> System.IndexOutOfRangeException: Index was outside the bounds of the array.
         at Internal.TypeSystem.ArrayMethod.GetCanonMethodTarget(CanonicalFormKind kind)

```
full log: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-48152-merge-304a3fe532ab482ca7/PayloadGroup0/console.88cdfe54.log?sv=2019-07-07&se=2021-03-03T22%3A44%3A29Z&sr=c&sp=rl&sig=oyTi3cN3NEeo8ng0oocOqdvb8kdokAjUMLrxnXZrzk8%3D

added in https://github.com/dotnet/runtime/pull/47400, @MichalStrehovsky could you please take a look if it is a test issue?

**Comment by MichalStrehovsky** [comment]
From the stack this looks like a crossgen2 issue.  I saw the same issue in NativeAOT. I assume Crossgen2 needs a fix along the lines of https://github.com/dotnet/runtimelab/pull/586 (the CorInfoImpl part). Array methods are weird and their canonical forms make things weirder.

**Comment by jkotas** [comment]
Fix is getting reverted in #48557
